src/smpi/smpi_shared.cpp

```diff
@@ -125,6 +127,7 @@ std::map<void*, shared_metadata_t> allocs_metadata;
 xbt_dict_t calls = nullptr;           /* Allocated on first use */
 #ifndef WIN32
 static int smpi_shared_malloc_bogusfile           = -1;
+static int smpi_shared_malloc_bogusfile_huge_page  = -1;
```

Comment:
When huge pages are used, there are still some "classical" calls to mmap without `MAP_HUGETLB` for areas smaller than a huge page (the start and the end of the blocks).
Apparently, you cannot call mmap without `MAP_HUGETLB` on a file opened in a `hugetlbfs` mount point. So you need two files in this case.